### PR TITLE
Normalize DB timestamp storage boundaries

### PR DESCRIPTION
**Pull request by hoonly01**
## Summary
- Add shared timestamp utilities for UTC-aware DB instants, KST wall-clock schedules, and legacy timestamp parsing.
- Store new alarm/user/Kakao/SMPA instant timestamps as UTC-aware ISO strings while preserving SMPA start/end as Seoul local schedule strings.
- Render admin timestamps in KST according to storage meaning and add regression coverage for UTC storage vs KST display behavior.

## Documentation scope
- README is intentionally unchanged per maintainer direction; technical storage details are documented in code/tests and this PR description instead.

## Tests
- `git diff --check`
- `PYTHONPYCACHEPREFIX="$(mktemp -d)" uv run python -m compileall -q app tests`
- `env -u RUN_LIVE_SMPA uv run pytest` — 173 passed, 2 skipped

Refs #119

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved timestamp accuracy and timezone consistency across user operations, message handling, and system maintenance tasks.
  * Fixed data integrity issues related to mutable default values in data models.

* **Refactor**
  * Standardized datetime parsing and timezone management throughout the application for consistent, reliable timekeeping.
  * Centralized time utility functions to enhance system reliability and reduce code duplication.

[Review Change Stack](https://app.coderabbit.ai/change-stack/hoonly01/kt-demo-alarm/pull/126?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: app/services/alarm_status_service.py [hunk 1]
```diff
@@ -252,17 +254,15 @@ def cleanup_old_tasks(days: int = 30) -> int:
...
             with get_db_connection() as db:
                 cursor = db.cursor()
                 cursor.execute(
                     """
                     DELETE FROM alarm_tasks 
-                    WHERE created_at < ?
+                    WHERE datetime(created_at) < datetime(?)
                     """,
                     (cutoff_date,)
                 )
```
[the remark this answers is not in the document]
Comment (hoonly01) [reply app/services/alarm_status_service.py]: Decision: implemented

Rationale: Must Fix로 판단했습니다. 기존 `alarm_tasks.created_at` row는 `datetime.now().isoformat()` 기반 naive KST 값일 수 있어서 SQLite `datetime()` 직접 비교는 retention cutoff 근처에서 실제 시점 기준 삭제 여부를 잘못 판단할 수 있습니다. Issue #119가 backfill을 제외하므로 cleanup 경계에서 legacy timezone 정책을 명시하는 쪽이 안전합니다.

Commit hash: 4f751c5

Concise summary of the change: `cleanup_old_tasks()`가 각 row의 `created_at`을 `parse_db_timestamp(..., naive_source_tz=KST)`로 파싱한 뒤 aware datetime끼리 비교하도록 변경했고, legacy naive KST row가 cutoff 기준으로 정리되는 회귀 테스트를 추가했습니다.

Validation performed: LSP diagnostics clean for `app/services/alarm_status_service.py` and `tests/test_alarm_status_service.py`; `env -u RUN_LIVE_SMPA uv run pytest tests/test_alarm_status_service.py tests/test_time_utils.py`; `git diff --check`; `PYTHONPYCACHEPREFIX="$(mktemp -d)" uv run python -m compileall -q app tests`; `env -u RUN_LIVE_SMPA uv run pytest` (174 passed, 2 skipped).

**Comment by hoonly01** [comment]
### Review comment assessment and decisions

| Original thread | Category | Decision | Rationale | Commit hash | Validation performed |
|---|---|---|---|---|---|
| CodeRabbit rate-limit notice (`issue comment 4476805570`) | Not Needed | not implemented | 리뷰 사용량 제한 안내이며 코드 변경 요청이 아닙니다. PR 코드의 정확성이나 Issue #119 구현 범위와 직접 관련된 수정 요구가 없습니다. | N/A | PR comments/reviews/review-comments inventory checked via GitHub API. |
| Copilot inline comment on `cleanup_old_tasks` legacy naive KST comparison (`review comment 3258197106`) | Must Fix | implemented | 기존 `alarm_tasks.created_at` row는 naive KST 문자열일 수 있으므로 SQLite `datetime()` 비교가 retention cutoff 근처에서 실제 시점 기준 삭제 여부를 잘못 판단할 수 있습니다. Backfill이 비범위라 cleanup 경계에서 명시적으로 legacy timezone을 적용해야 합니다. | 4f751c5 | LSP diagnostics clean for touched files; `env -u RUN_LIVE_SMPA uv run pytest tests/test_alarm_status_service.py tests/test_time_utils.py`; `git diff --check`; `PYTHONPYCACHEPREFIX="$(mktemp -d)" uv run python -m compileall -q app tests`; `env -u RUN_LIVE_SMPA uv run pytest` (174 passed, 2 skipped). |

Implemented summary for `4f751c5`: `cleanup_old_tasks()` now parses both UTC-aware rows and legacy naive KST rows into aware datetimes before retention comparison, and a regression test covers the legacy KST boundary case.

File: app/utils/time_utils.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    if parsed.tzinfo is None:
+        return parsed.replace(tzinfo=naive_source_tz)
+
+    return parsed
+
+
+def _datetime_from_epoch(value: int | float) -> datetime | None:
+    timestamp = float(value)
+    if timestamp > EPOCH_MILLISECONDS_THRESHOLD:
+        timestamp /= 1000.0
```
[the remark this answers is not in the document]
Comment (hoonly01) [reply app/utils/time_utils.py]: Decision: implemented

Rationale: Must Fix로 판단했습니다. `EPOCH_MILLISECONDS_THRESHOLD`와 정확히 같은 값은 초 단위로 해석하면 `datetime.fromtimestamp()` 범위를 벗어나거나 잘못된 시점이 되므로, 이 경계값은 ms epoch로 처리해야 합니다.

Commit hash: ef67af9

Concise summary of the change: `_datetime_from_epoch()`의 threshold 비교를 `>`에서 `>=`로 바꾸고, 경계값을 ms로 파싱하는 회귀 테스트를 추가했습니다.

Validation performed: LSP diagnostics clean for `app/utils/time_utils.py`, `tests/test_time_utils.py`, and `tests/test_api_admin.py`; `env -u RUN_LIVE_SMPA uv run pytest tests/test_time_utils.py tests/test_api_admin.py` (38 passed); `git diff --check`; `PYTHONPYCACHEPREFIX="$(mktemp -d)" uv run python -m compileall -q app tests`; `env -u RUN_LIVE_SMPA uv run pytest` (175 passed, 2 skipped).

File: tests/test_api_admin.py [hunk 3]
```diff
@@ -318,6 +334,9 @@ def test_admin_dashboard_displays_times_in_kst(test_client, monkeypatch):
     assert "2026-05-15 20:00:00 KST" not in response.text
     assert "kst-user" in response.text
     assert "2026-05-20 KST" in response.text
+    assert "utc-user" in response.text
+    assert "2026-05-20 09:30:00 KST" in response.text
+    assert "2026-05-20 09:40:00 KST" in response.text
```
[the remark this answers is not in the document]
Comment (hoonly01) [reply tests/test_api_admin.py]: Decision: implemented

Rationale: Optional이지만 reasonable/low-risk로 판단했습니다. 기존 assertion은 같은 응답의 다른 row에 같은 timestamp가 있어도 통과할 수 있어, `utc-user` row 자체의 변환을 검증하도록 좁히는 것이 현재 범위 안의 테스트 품질 개선입니다.

Commit hash: ef67af9

Concise summary of the change: rendered HTML에서 `utc-user`가 포함된 `<tr>...</tr>` 블록을 찾고, 그 row 안에 `2026-05-20 09:30:00 KST`와 `2026-05-20 09:40:00 KST`가 있는지 검증하도록 수정했습니다.

Validation performed: LSP diagnostics clean for `app/utils/time_utils.py`, `tests/test_time_utils.py`, and `tests/test_api_admin.py`; `env -u RUN_LIVE_SMPA uv run pytest tests/test_time_utils.py tests/test_api_admin.py` (38 passed); `git diff --check`; `PYTHONPYCACHEPREFIX="$(mktemp -d)" uv run python -m compileall -q app tests`; `env -u RUN_LIVE_SMPA uv run pytest` (175 passed, 2 skipped).

**Comment by hoonly01** [comment]
### 추가 CodeRabbit review 처리 결과

| Original thread/comment | Category | Decision | Rationale | Commit hash | Validation performed |
|---|---|---|---|---|---|
| `app/utils/time_utils.py` epoch-ms threshold boundary (`review comment 3258281094`) | Must Fix | implemented | `EPOCH_MILLISECONDS_THRESHOLD`와 같은 값은 초 단위로 해석하면 잘못된/범위 밖 timestamp가 되므로 ms epoch로 처리해야 하는 실제 파싱 버그입니다. | ef67af9 | LSP diagnostics clean; `env -u RUN_LIVE_SMPA uv run pytest tests/test_time_utils.py tests/test_api_admin.py` (38 passed); `git diff --check`; `compileall`; full `env -u RUN_LIVE_SMPA uv run pytest` (175 passed, 2 skipped). |
| `tests/test_api_admin.py` utc-user timestamp assertion locality (`review comment 3258281138`) | Optional | implemented | 테스트 품질 개선이며 현재 범위 안에서 저위험입니다. `utc-user` row 자체에 변환된 KST timestamp가 있는지 검증하도록 좁혔습니다. | ef67af9 | Same validation as above. |
| `app/routers/admin.py` alarm display formatter nitpick in review body (`review 4309479385`) | Not Needed | not implemented | 제안대로 `_format_utc_timestamp_as_kst`로 바꾸면 legacy naive `alarm_tasks` row를 UTC naive로 해석해 기존 KST-local 표시 호환성을 깨며, 현재 `_format_kst_local_datetime`은 aware 값은 offset 기반으로 KST 변환하고 naive 값만 KST로 해석합니다. 따라서 제안은 현재 legacy 정책과 충돌합니다. | N/A | Current behavior covered by full test suite and prior legacy timestamp policy checks. |
| CodeRabbit docstring coverage warning / finishing touch (`issue comment 4476805570`) | Optional | deferred | repo-wide docstring coverage 80% 달성은 광범위한 문서화 작업이며 Issue #119 timestamp 저장 경계 수정 범위를 벗어납니다. 이번 PR에서는 기능/회귀 안정성에 필요한 범위만 처리했습니다. | N/A | Not applicable. |
